homework_1/task3/main.go

```diff
@@ -0,0 +1,199 @@
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"io/ioutil"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/sethgrid/pester"
+)
+
+const pokemonURL = "https://pokeapi.co/api/v2/pokemon/"
+
+func linearBackoff(retry int) time.Duration {
+	return time.Duration(retry) * time.Second
+}
+
+//this struct is used in pokemonLocalEncounters to get the location name and
+//in when finding pokemons name if not given
+type nameGetter struct {
+	Name 		string	  `json:"name"`
+}
+
+type idGetter struct {
+	Id 			int 	 `json:"id"`
+}
+
+type pokemonLocalEncounters struct {
+	LocalEncounterArea nameGetter	`json:"location_area"`
```

Comment:
We've discussed this in person; I'd prefer not reusing `nameGetter`, but honestly this is also fine.

Reply:
Yes, as per our discussion, I have added new struct `locationAreaName` and changed `nameGetter` to `pokemonName` as it describes its purpose in a much clearer way.